File: ios/RandomModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RandomModule, NSObject)
    RCT_EXTERN_METHOD(
        getRandomColor:
        (RCTPromiseResolveBlock)resolve
        rejecter:(RCTPromiseRejectBlock)reject
    )
@end

File: ios/RandomModule.swift
import Foundation

@objc(RandomModule)
class RandomModule: NSObject {
    static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc
    func getRandomColor(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
        resolve([
            Int.random(in:1...255),
            Int.random(in:1...255),
            Int.random(in:1...255)
        ])
    }
}
